public/components/trace_analytics/components/traces/span_detail_table.tsx

```diff
@@ -261,37 +252,246 @@ export function SpanDetailTable(props: SpanDetailTableProps) {
+        .map(({ id }) => id)
+    );
+  }, [columns, props.hiddenColumns]);
+
+  const [fullScreenMode, setFullScreenMode] = useState(false);
+  const openFullScreenModal = () => setFullScreenMode(true);
+  const closeFullScreenModal = () => setFullScreenMode(false);
+
+  const renderCellValue = useCallback(
+    (params) =>
+      renderCommonCellValue({
+        ...params,
+        items,
+        props,
+        expandedRows,
+        toggleRowExpansion: (id) => {
+          setExpandedRows((prev) => {
+            const newSet = new Set(prev);
+            if (newSet.has(id)) {
+              newSet.delete(id);
+            } else {
+              newSet.add(id);
+            }
+            return newSet;
+          });
+        },
+        flattenedItems,
+        indentationFactor: 20,
+        fullScreenMode,
+      }),
+    [items, expandedRows, props, flattenedItems, fullScreenMode]
+  );
+
+  const gatherAllSpanIds = (spans: Span[]): Set<string> => {
+    const allSpanIds = new Set<string>();
+
+    spans.forEach((span) => {
+      allSpanIds.add(span.spanId);
+
+      if (span.children && span.children.length > 0) {
+        const childSpanIds = gatherAllSpanIds(span.children);
+        childSpanIds.forEach((id) => allSpanIds.add(id));
+      }
+    });
+
+    return allSpanIds;
+  };
+
+  const expandAllRows = () => {
+    const allExpandedIds = gatherAllSpanIds(items);
+    setExpandedRows(allExpandedIds);
+  };
+
+  const collapseAllRows = () => {
+    setExpandedRows(new Set());
+  };
+
+  const toolbarButtons = [
+    <EuiButtonEmpty
+      size="xs"
+      onClick={fullScreenMode ? closeFullScreenModal : openFullScreenModal}
+      key="fullScreen"
+      color="text"
+      iconType={fullScreenMode ? 'cross' : 'fullScreen'}
+      data-test-subj="fullScreenButton"
+    >
+      {fullScreenMode ? 'Exit full screen' : 'Full screen'}
+    </EuiButtonEmpty>,
+    <EuiButtonEmpty
+      size="xs"
+      onClick={expandAllRows}
+      key="expandAll"
+      color="text"
+      iconType="expand"
+    >
+      Expand all
+    </EuiButtonEmpty>,
+    <EuiButtonEmpty
+      size="xs"
+      onClick={collapseAllRows}
+      key="collapseAll"
+      color="text"
+      iconType="minimize"
+    >
+      Collapse all
+    </EuiButtonEmpty>,
+  ];
+
+  return (
+    <>
+      <FullScreenWrapper isFullScreen={fullScreenMode} onClose={closeFullScreenModal}>
+        <EuiDataGrid
+          aria-labelledby="span-detail-data-grid"
+          columns={columns}
+          columnVisibility={{ visibleColumns, setVisibleColumns }}
+          rowCount={flattenedItems.length}
+          renderCellValue={renderCellValue}
+          toolbarVisibility={{
+            showColumnSelector: true,
+            showSortSelector: true,
+            showFullScreenSelector: false,
+            additionalControls: toolbarButtons,
+          }}
+          style={{ height: fullScreenMode ? '100%' : '500px', overflowY: 'auto' }}
```

Comment:
Why is the datagrid restricted to 500px when not in fullscreen? Does this change on a bigger screen?

Reply:
Auto can't be used as the window is initially rendered with the tree collapsed. So the window becomes roughly 50 pixels that is scrollable. 500pixels matches the limit of the "Span Tree" and keeps the window size for that panel consistent.